stuff/cron/plagiarism_detector.py

```diff
@@ -45,111 +68,210 @@
                                             (SELECT p.`contest_id` 
                                             FROM `Plagiarisms` as p);
                                 """
-GET_CONTEST_SUBMISSION_IDS= """ SELECT c.contest_id, s.submission_id, s.problemset_id,
+
+
+class Contest(TypedDict):
+    contest_id: int
+    problemset_id: int
+
+
+GET_CONTEST_SUBMISSION_IDS = """ SELECT c.contest_id, s.submission_id, s.problemset_id,
                                 s.problem_id, s.verdict, s.guid, s.language
                                 FROM Submissions as s 
                                 INNER JOIN Contests c ON c.problemset_id = s.problemset_id 
                                 WHERE c.contest_id = %s;
                             """
+
+
+class Submission(TypedDict):
+    contest_id: int
+    submission_id: int
+    problemset_id: int
+    problem_id: int
+    verdict: str
+    guid: str
+    language: str
+
+
 INSERT_INTO_PLAGIARISMS = """  
                                 INSERT INTO `Plagiarisms`
                                 (`contest_id`, `score_1`, `score_2` , `submission_id_1` , `submission_id_2`, `contents`)
                                 VALUES
                                 (%s, %s, %s, %s, %s, %s)
                             """
 
-#Constants
+# Constants
 START_RED = "<span class='highlight-red'>"
 START_GREEN = "<span class='highlight-green'>"
 END = "</span>"
 
-SubmissionDownloader = Callable[[str, str], None]
+SubmissionDownloader = Callable[[str, str, str], None]
+
 
 class S3SubmissionDownloader:
     ''' A SubmissionDownloader that can download files from an S3 bucket.  '''
 
     def __init__(self, bucket_name: str = 'omegaup-backup') -> None:
         self._bucket = boto3.client('s3').Bucket(bucket_name)
-    
-    def __call__(self, guid: str, destination_path: str) -> None:
-        self._bucket.download_file(f'omegaup/submissions/{guid[:2]}/{guid[2:]}', destination_path)
-    
+
+    def __call__(self, guid: str, destination_dir: str, language: str) -> None:
+        self._bucket.download_file(
+            f'omegaup/submissions/{guid[:2]}/{guid[2:]}',
+            os.path.join(destination_dir, f'{guid}.{language}'))
+
+
 class LocalSubmissionDownloader:
 
     def __init__(self, dir: str) -> None:
-        self._dir = dir
-    
-    def __call__(self, guid: str, destination_path: str) -> None:
-        
-        shutil.copyfile(os.path.join(self._dir, "test1.cpp"), destination_path)
+        self._dir = os.path.join(SUBMISSION_PATH, dir)
```

Comment:
```suggestion
        self._dir = os.path.join(self._dir, dir)
```

let's avoid hardcoded paths